src/Elastic.Apm/Helpers/StacktraceHelper.cs

```diff
@@ -28,14 +28,15 @@ internal static List<CapturedStackFrame> GenerateApmStackTrace(StackFrame[] fram
 			try
 			{
 				retVal.AddRange(from item in frames
+					let fileName = item?.GetMethod()
+						?.ReflectedType?.FullName //see: https://github.com/elastic/apm-agent-dotnet/pull/240#discussion_r289619196
+					let functionName = GetRealMethodName(item?.GetMethod())
 					select new CapturedStackFrame
 					{
-						Function = GetRealMethodName(item?.GetMethod()),
-						FileName =
-							item?.GetMethod()
-								?.ReflectedType?.FullName, //see: https://github.com/elastic/apm-agent-dotnet/pull/240#discussion_r289619196
+						Function = functionName,
+						FileName = string.IsNullOrWhiteSpace(fileName)? "unknown" : fileName,
```

Comment:
Do we have example when method type name is not available? Can we use something more useful than "unknown"? Maybe we can fallback on some other property from the stack frame info.

Reply:
I think item.GetFileName() can work here

Reply:
> Do we have example when method type name is not available?

Yes, run the `/home/index` in the ASP.NET Core sample application from master... the outgoing http request (where we capture the stacktrace) will be rejected, because the FileName remains `null`. It's a generated lambda. 

> I think item.GetFileName() can work here
Wouldn't that be very confusing? Also I think in that case `GetFileName` also returns `null`, but I haven't tested this, it's just a guess: if there is no `?.ReflectedType` maybe there is no filename either.

Reply:
Yes, GetFileName will not work for that 
![image](https://user-images.githubusercontent.com/3065048/59135670-7dc30080-8988-11e9-8461-239c3dcf5ad9.png)

Reply:
Thanks for trying @skynet2! I was about to do that.. appreciate the info.

Reply:
Do you think that it makes sense to include such data in stack trace? 
Without fileName, it makes no sense to have data in apm. As i can remember, previously, there was a check to check if fileName is not empty, and only then include it as a stackframe

Reply:
So.. to answer the question:

> Can we use something more useful than "unknown"? 

I don't think so... it's literally unknown. That's some runtime generated stuff, which has no type name and also no filename. Also, I think in msil it's possible to have global functions without having a class.

Reply:
> Do you think that it makes sense to include such data in stack trace?
> Without fileName, it makes no sense to have data in apm. As i can remember, previously, there was a check to check if fileName is not empty, and only then include it as a stackframe

Hmm, yeah, now that you say this, I remember that we dropped those things originally. I also think that this doesn't really add much value, on the other hand cutting out things from a stacktrace does not sound very good. What if you e.g. also have a memory dump and you want to search for similar callstack in that? If we skip frames we make that harder. But I'm not 100% sure.

Reply:
hm, I don't remember the code where I dropped it, so probably it was not me. Don`t have git log currently :) 
Working with a memory dump from prod, IMHO, that's not a common scenario

Reply:
I just wanted to make sure we cannot find anything better, at least for the cases of which we know. If there isn't anything better then let's go with unknown. I would just make sure it cannot be confused with a real class name - maybe something like `"<unknown>"`?

Reply:
> hm, I don't remember the code where I dropped it, so probably it was not me. Don`t have git log currently :)

I think it was me, my original code skipped those. :) 

> Working with a memory dump from prod, IMHO, that's not a common scenario 
Sure, it's not common. My point is that if we skip frames then we don't really show "THE" stacktrace, we just show a modified stacktrace... it's not the real stacktrace, and I think this can be misleading.

Reply:
>  maybe something like "<unknown>"? 

Sounds good to me.

Reply:
> > maybe something like `"<unknown>"`?
> 
> Sounds good to me.

@skynet2 also ok with this?

Reply:
yep, let`s try that solution, thanks, @gregkalapos

Reply:
Nit: I thought about even better alternative - how about `"N/A"`? You guys choose :)

Reply:
Ok, let's go with `N/A`.